new_course/1_programming_fundementals/methods.md

```diff
@@ -1,62 +1,268 @@
 # Methods
 Methods, also known as functions, are one of the foundational building blocks in programming. Almost every language implements some way of making methods.
 
-You will often find yourself writing the same code in different places in your program. Wouldn't it be great if there was a way to reuse the same code over and over again without having to write it all out each time?
+You will often find yourself writing the same code in different places in your program. Wouldn't it be great if there was a way to reuse the same code over and over again without having to write it all out each time? This is what methods are for. They allow you to wrap sections of your code in a name, which you can then use where you need that code to be run in your program, and as many times are you need.
 
-This is what methods are for. They allow you to wrap code in a name which you
-can then use where you need that code to be run in your programs.
+This concept allows for what programmers refer to as the *D.R.Y.* approach to programming *-Don't Repeat Yourself-*. Recycling parts of code that would otherwise need to be repeated, often numerous times in several places, *can* make your program more readable and more manageable.
 
 In this lesson we are going to deconstruct what methods are, their behaviour, and how they are used.
 
 ## Learning Outcomes
 *Look through these now and then use them to test yourself after doing the assignment*
 
 * You understand how to use Ruby's built in methods on objects
-* You know how to create your own custom methods
-* You know how to *call* your own methods
-* You understand the difference between a explicit and implicit return in your methods
-* You understand the difference between `puts` and `return`
+* How would you create your own custom method?
+* How would you *call* your own method?
+* How do you pass variables to your method?
+* How do you define default parameters for your method?
+* What is the difference between an *explicit* and *implicit* return in your methods?
+* What is the difference between `puts` and `return`?
+* How do you chain multiple methods together?
+* Give an example of a valid method name and an invalid method name.
+* What does "snake case" naming mean?
+* What are some of Ruby's reserved words?
+* What do you call a method that returns `true` or `false`? How are they named?
+* What do bang methods do? How are they conventionally named?
 
 ## Ruby's Built in Methods
-* explain what they are, you've already been using them.
+
+One of Ruby's great advantages for new programmers is the large number of built-in methods it includes. You've been using many of them already, probably without even realizing it. Over the course of your learning so far, you have modified strings and other objects in various ways. Each of these operations you have performed has been a method. These are most often applied by adding `.method_name` after the object you are modifying.
```

Comment:
>Each of these operations you have performed has been a method.

This doesn't read well to me and may need to be slightly changed.

>These are most often applied by adding `.method_name` after the object you are modifying.

Not all methods are used to modify an object.